R/windsorize.R

```diff
@@ -1,10 +1,36 @@
 #' Windsorize
 #'
-#' Do some windsorization.
+#' Its purposes is to eliminate outliers in a following way. Values of (0.5 +- p/2)th quantiles are calculated and all
+#' values above(below) those quantiles are replaced by the quantiles.
+#' @param x a numeric vector
+#' @param p quantile
+#' @return Windsorized  vector \code{x}
+#' @examples
+#' example_vector=c(-1000,1,2,3,4,5,6,7,8,9,1000)
+#' windsorize(example_vector, 0.9)
+#' 
+#' example_vector=rnorm(100)
+#' windsorize(example_vector, 0.9)
 #' @export
+#' @import stats
+
 windsorize <- function(x, p = .90) {
-  q <- quantile(x, p)
-  x[x >= q] <- q
-  x
+  
+  if(is.null(x)) {stop("Input vector cannot be NULL.")}
+  if(any(is.na(x))) {stop("There should be no NA's in input vector.")}
```

Comment:
1. Why  are no `NA`'s allowed? What could be the workaround here to still windsorize and keep `NA`s?
2. Typo: There should be no NA's in **the** input...